src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/CodeGeneration/MethodGenerator.cs

```diff
@@ -228,12 +227,24 @@ private static SyntaxList<TypeParameterConstraintClauseSyntax> GenerateDefaultCo
                 if (!referencedTypeParameters.Contains(typeParameter))
                     continue;
 
-                var constraint = typeParameter switch
+                TypeParameterConstraintSyntax constraint;
+
+                if (typeParameter.HasReferenceTypeConstraint)
+                {
+                    constraint = s_classConstraint;
+                }
+                else if (typeParameter.HasValueTypeConstraint)
+                {
+                    constraint = s_structConstraint;
+                }
+                else if (typeParameter.ConstraintTypes.Any(t => t.IsReferenceType))
+                {
+                    constraint = s_classConstraint;
+                }
```

Comment:
is the order of these checks important?  if not, i'd merge this with the top check.